airbyte-integrations/connectors/source-s3/source_s3/v4/stream_reader.py

```diff
@@ -68,25 +69,30 @@ def get_matching_files(self, globs: List[str], prefix: Optional[str], logger: lo
         total_n_keys = 0
 
         try:
-            if prefixes:
-                for prefix in prefixes:
-                    for remote_file in self._page(s3, globs, self.config.bucket, prefix, seen, logger):
-                        total_n_keys += 1
-                        yield remote_file
-            else:
-                for remote_file in self._page(s3, globs, self.config.bucket, None, seen, logger):
+            for current_prefix in (prefixes if prefixes else [None]):
+                for remote_file in self._page(s3, globs, self.config.bucket, current_prefix, seen, logger):
                     total_n_keys += 1
                     yield remote_file
 
             logger.info(f"Finished listing objects from S3. Found {total_n_keys} objects total ({len(seen)} unique objects).")
+        except ClientError as exc:
+            if exc.response["Error"]["Code"] == "NoSuchBucket":
+                error = f"The bucket {self.config.bucket} does not exist."
+            else:
+                error = FileBasedSourceError.ERROR_LISTING_FILES
+            self.raise_error_listing_files(error, globs, exc)
         except Exception as exc:
-            raise ErrorListingFiles(
-                FileBasedSourceError.ERROR_LISTING_FILES,
-                source="s3",
-                bucket=self.config.bucket,
-                globs=globs,
-                endpoint=self.config.endpoint,
-            ) from exc
+            self.raise_error_listing_files(FileBasedSourceError.ERROR_LISTING_FILES, globs, exc)
+
+    def raise_error_listing_files(self, error_message: Union[str, FileBasedSourceError], globs: List[str], exc: Optional[Exception] = None):
```

Comment:
Why do we need a helper method for this?

Reply:
I used the helper method to reduce repetitive error-raising code.